src/Storages/MergeTree/ReplicatedMergeMutateTaskBase.h

```diff
@@ -0,0 +1,77 @@
+#pragma once
+
+#include <common/logger_useful.h>
+
+#include <Storages/MergeTree/IExecutableTask.h>
+#include <Storages/MergeTree/ReplicatedMergeTreeQueue.h>
+
+namespace DB
+{
+
+class StorageReplicatedMergeTree;
+
+/**
+ * This is used as a base of MergeFromLogEntryTask and MutateFromLogEntryTaskBase
+ */
+class ReplicatedMergeMutateTaskBase : public IExecutableTask
+{
+public:
+    template <class Callback>
+    ReplicatedMergeMutateTaskBase(
+        Poco::Logger * log_,
+        StorageReplicatedMergeTree & storage_,
+        ReplicatedMergeTreeQueue::SelectedEntryPtr & selected_entry_,
+        Callback && task_result_callback_)
```

Comment:
Please, at least add a comment that `task_result_callback` is called on task finish, with arg=true if execution was successful.